frontend/src/pages/_app.tsx

```diff
@@ -135,10 +135,16 @@ const MenuContent: React.FC<{ hasAdminPermission: boolean }> = ({ hasAdminPermis
 const RoutingMiddleware: React.FC<{ children: ReactNode }> = ({ children }) => {
   const router = useRouter();
   const hasMemberPermissions = useHasMemberPermission();
+  const hasAdminPermission = useHasAdminPermission();
 
-  if (!router.pathname.startsWith('/applicant') && !hasMemberPermissions) {
-    router.push('/applicant');
-  }
-
+  useEffect(() => {
+    if (!router.pathname.startsWith('/applicant') && !hasMemberPermissions) {
+      router.push('/applicant');
+      return;
+    }
+    if (router.pathname.startsWith('/admin') && !hasAdminPermission) {
+      router.push('/');
```

Comment:
```suggestion
    if (router.pathname !== '/') router.push('/');
```
Also maybe adding the return as well to keep consistency.